packages/expo-location/android/src/main/java/expo/modules/location/LocationModule.java

```diff
@@ -463,7 +463,12 @@ public void hasServicesEnabledAsync(final Promise promise) {
 
   @ExpoMethod
   public void startLocationUpdatesAsync(String taskName, Map<String, Object> options, final Promise promise) {
-    if (isMissingBackgroundPermissions()) {
+    boolean useForegroundService = LocationTaskConsumer.useForegroundService(options);
```

Comment:
Renaming to `shouldUseForegroundService`:

```suggestion
    boolean shouldUseForegroundService = LocationTaskConsumer.shouldUseForegroundService(options);
```